### PR TITLE
Separate rosdep collection from sysroot creation

### PR DESCRIPTION
**Pull request by emersonknapp**
Fixes https://github.com/ros-tooling/cross_compile/issues/106

Creates a separate container, that runs natively, and outputs a script for installing workspace dependencies.

If the script has not changed (no dependencies have changed), the docker cache is used and the build can begin immediately. This makes the tool much more usable by skipping the very slow rosdep installation step whenever a source file is changed.

Note that docker-based tests are not run on our macOS Github Action hosts, because they will not install docker on those due to licensing issues.

### CODE REVIEW

File: ros_cross_compile/dependencies.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    :param platform: The name of the image produced by `build_rosdep_image`
+    :param workspace: Absolute path to the colcon source workspace.
+    :param custom_script: Optional absolute path of a script that does custom setup for rosdep
+    :param custom_data_dir: Optional absolute path of a directory containing custom data for setup
+    :return None
+    """
+    image_name = 'ros_cross_compile:rosdep'
+    logger.info('Building rosdep collector image: %s', image_name)
+    docker_client.build_image(
+        dockerfile_name='rosdep.Dockerfile',
```
Comment (piraka9011) [inline ros_cross_compile/dependencies.py]: Should this be something like `Dockerfile.rosdep` instead?

File: ros_cross_compile/dependencies.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    )
+
+    logger.info('Running rosdep collector image on workspace {}'.format(workspace))
+    volumes = {
+        workspace: '/ws',
+    }
+    if custom_script:
+        volumes[custom_script] = CUSTOM_SETUP
+    if custom_data_dir:
+        volumes[custom_data_dir] = '/usercustom/custom-data'
```
Comment (piraka9011) [inline ros_cross_compile/dependencies.py]: Does this need to be a named constant? looks random

File: ros_cross_compile/dependencies.py (hunk 1 above)
Comment (emersonknapp) [reply ros_cross_compile/dependencies.py]: i prefer the pattern of "the extension tells you what kind of file it is". and we're already using it for `sysroot.Dockerfile`

File: test/test_dependencies.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+@pytest.mark.skipif(is_mac(), reason=NO_MAC_REASON)
+def test_dummy_ros2_pkg(tmpdir):
+    ws = Path(str(tmpdir))
+    pkg_xml = ws / 'src' / 'dummy' / 'package.xml'
+    out_script = ws / DEPENDENCY_SCRIPT_SUBPATH
+    pkg_xml.parent.mkdir(parents=True)
+    pkg_xml.write_text(RCLCPP_PKG_XML)
+
+    client = DockerClient()
+    platform = Platform(arch='aarch64', os_name='ubuntu', ros_distro='dashing')
```
Comment (zmichaels11) [inline test/test_dependencies.py]: I think we should have a TODO comment in this test to switch to native arch if supported

**Pull request merged**